WordPress/gutenberg
# ToggleGroupControl: rewrite backdrop animation with framer-motion's shared layout animations

## What?


Completely rewrite the `ToggleGroupControl` 's animated backdrop using `framer-motion`'s [shared layout animations](https://www.framer.com/docs/layout-group/).

Supersedes #40107 

Fixes https://github.com/WordPress/gutenberg/issues/37506

## Why?


Current implementation is complex, difficult to read, imperative (based off arbitrary timeouts), and [buggy](https://github.com/WordPress/gutenberg/issues/37506).


## How?


Main changes are:

- deleted the `ToggleGroupControlBackdrop`, and the imperative logic that came with it
- re-implemented the backdrop using `framer-motion`'s shared layout animations:
    - the main way this works is by using the same value for the `layoutId` attribute to tag the components that participate to the same shared layout animation
    - each `ToggleGroupControl` also defines its own "namespace" for the shared layout animations via the `LayoutGroup` component, in order to avoid conflicts between different instances of the component
- cleanup up some styles and removed some unused variables
- memoized some classnames

## Testing Instructions





Make sure that the backdrop animation works as expected:
  - in Storybook (check all of the different examples for `ToggleGroupControl`)
  - in the block editor (in particular, test for the scenario highlighted in #37506)

## Screenshots or screencast 

https://user-images.githubusercontent.com/1083581/162973588-26f77da2-4c4e-4c2f-a953-ef7ab81d3198.mp4
